raptor/playback_tools/mitmproxy/mitmproxy.py

```diff
@@ -0,0 +1,309 @@
+                if self.mitmproxy_rel_bin is None:
+                    self.fatal("Aborting: mitmproxy_release_bin_osx not found in talos.json")
+                self.download_mitmproxy_binary(_platform)
+            else:
+                self.info("Not downloading mitmproxy rel binary because no-download was specified")
+        self.info('The mitmdump macosx binary is found at: %s' % self.mitmdump)
+    self.run_command([self.mitmdump, '--version'], env=self.query_env())
+
+
+def query_mitmproxy_rel_bin(self, platform):
+    """Mitmproxy requires external playback archives to be downloaded and extracted"""
+    if self.mitmproxy_rel_bin:
+        return self.mitmproxy_rel_bin
+    if self.query_talos_json_config() and self.suite is not None:
+        config_key = "mitmproxy_release_bin_" + platform
+        self.mitmproxy_rel_bin = self.talos_json_config['suites'][self.suite].get(config_key, False)
+        return self.mitmproxy_rel_bin
+
+
+def download_mitmproxy_binary(self, platform):
+    """Download the mitmproxy release binary from tooltool"""
+    self.info("Downloading the mitmproxy release binary using tooltool")
+    dest = os.path.join(self.talos_path, 'talos', 'mitmproxy')
+    _manifest = "mitmproxy-rel-bin-%s.manifest" % platform
+    manifest_file = os.path.join(self.talos_path, 'talos', 'mitmproxy', _manifest)
+
+    if platform in ['osx', 'linux64']:
+        self.tooltool_fetch(
+            manifest_file,
+            output_dir=dest,
+            cache=self.config.get('tooltool_cache')
+        )
+
+        archive = os.path.join(dest, self.mitmproxy_rel_bin)
+        tar = self.query_exe('tar')
+        unzip_cmd = [tar, '-xvzf', archive, '-C', dest]
+        self.run_command(unzip_cmd, halt_on_failure=True)
+
+
+def query_mitmproxy_recording_set(self):
+    """Mitmproxy requires external playback archives to be downloaded and extracted"""
+    if self.mitmproxy_recording_set:
+        return self.mitmproxy_recording_set
+    if self.query_talos_json_config() and self.suite is not None:
+        self.mitmproxy_recording_set = self.talos_json_config['suites'][self.suite].get('mitmproxy_recording_set', False)
+        return self.mitmproxy_recording_set
+
+
+def download_mitmproxy_recording_set(self):
+    """Download the set of mitmproxy recording files that will be played back"""
+    self.info("Downloading the mitmproxy recording set using tooltool")
+    dest = os.path.join(self.talos_path, 'talos', 'mitmproxy')
+    manifest_file = os.path.join(self.talos_path, 'talos', 'mitmproxy', 'mitmproxy-playback-set.manifest')
+    self.tooltool_fetch(
+        manifest_file,
+        output_dir=dest,
+        cache=self.config.get('tooltool_cache')
+    )
+    archive = os.path.join(dest, self.mitmproxy_recording_set)
+    unzip = self.query_exe('unzip')
+    unzip_cmd = [unzip, '-q', '-o', archive, '-d', dest]
+    self.run_command(unzip_cmd, halt_on_failure=True)
+
+
+class Mitmproxy(object):
```

Comment:
I'd push for everything to be a module level method, or for everything to be encapsulated by this class. A mix of both could lead to some weirdness.